server/routes/probationPractitionerReferrals/myCasesPresenter.ts

```diff
@@ -0,0 +1,59 @@
+import { SentReferral, ServiceCategory } from '../../services/interventionsService'
+import PresenterUtils, { SortableTableHeaders, SortableTableRow } from '../../utils/presenterUtils'
+
+export default class MyCasesPresenter {
+  constructor(private readonly sentReferrals: SentReferral[], private readonly serviceCategories: ServiceCategory[]) {}
+
+  readonly tableHeadings: SortableTableHeaders = [
+    { text: 'Referral', sort: 'none' },
+    { text: 'Service user', sort: 'ascending' },
+    { text: 'Service Category', sort: 'none' },
+    { text: 'Provider', sort: 'none' },
+    { text: 'Caseworker', sort: 'none' },
+    { text: 'Action', sort: 'none' },
+  ]
+
+  readonly tableRows: SortableTableRow[] = this.sentReferrals.map(referral => {
+    const { serviceCategoryId } = referral.referral
+    const serviceCategory = this.serviceCategories.find(aServiceCategory => aServiceCategory.id === serviceCategoryId)
```

Comment:
this isn't DRY, and i don't like that the caller has to be aware that it needs just the service categories that are present in the list of referrals. would like to find a better way to do this.

Reply:
hmm, I wonder if this calls for a refactor of the referral object to include the service category name as well as the ID now? Not ideal though...